### PR TITLE
Homework 4/bet acceptance

### PR DESCRIPTION
**Pull request by jmilic1**
Part of homework 4 containing Bet Acceptance Api

For testing you can use script located at homework_3/03_project/script.sh to populate and initialize the database following with running Controller component.

Example requests:
POST http://127.0.0.1:8083/bet
```json
{
    "CustomerId": "6e90ed09-5815-4b93-72a4-a8048ead4aa5",
    "SelectionId": "bad76754-7c4c-4b93-50cb-14d17c51c372",
    "SelectionCoefficient": 6.30,
    "Payment": 44.195
}
```

### CODE REVIEW

File: homework_4/betacceptance/internal/api/controllers/validators/betRequestValidator.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package validators
+
+import "github.com/superbet-group/code-cadets-2021/homework_4/betacceptance/internal/api/controllers/models"
+
+const paymentFromInclusive = 2.0
+const paymentToInclusive = 100.0
+const coefficientToInclusive = 10.0
```
Comment (vpintaric-axilis) [inline homework_4/betacceptance/internal/api/controllers/validators/betRequestValidator.go]: Validation should be configurable. What if I want to deploy this service to two different countries with different laws? I would need to deploy 2 different versions of this service with different constants baked in.

File: homework_4/betacceptance/go.mod [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module github.com/superbet-group/code-cadets-2021/homework_4/betacceptance
+
+go 1.16
+
+require (
+	github.com/gin-gonic/gin v1.7.2
+	github.com/kelseyhightower/envconfig v1.4.0
+	github.com/nu7hatch/gouuid v0.0.0-20131221200532-179d4d0c4d8d
```
Comment (vpintaric-axilis) [inline homework_4/betacceptance/go.mod:8]: This is okay but there's https://github.com/google/uuid which:
* is more popular (more popular -> more users -> more bug reports/complaints/feature requests etc... -> more bugs fixed/features added etc... -> better library (most of the time))
* more up-to-date (last commit a couple of months ago while the nu7hatch one has last commit from 2013)
* uses go.mod so you can actually select a specific version and you don't need to use a commit hash

Again - no need to change it, it should get the work done. Just keep this in mind for future reference when selecting a library to use.

File: homework_4/betacceptance/internal/api/controllers/validators/betRequestValidator.go (hunk 1 above)
Comment (jmilic1) [reply homework_4/betacceptance/internal/api/controllers/validators/betRequestValidator.go]: I added extra parameters to .env config which the API reads while bootstrapping itself and assigns those values to Validator.
I'm not sure if these parameters should be defined there though.

Comment (vpintaric-axilis) [reply homework_4/betacceptance/internal/api/controllers/validators/betRequestValidator.go]: > I'm not sure if these parameters should be defined there though.

It's the exact place where those parameters should be defined (at least for this homework).

Of course there are multiple ways to handle configuration (or parts of the configuration) depending on what you need. For example:
* Occasionally - will it never change and will it always be the same wherever the app needs to be deployed? Simply use hard-coded constants.
* Most often - will it rarely change, should it be different across deployments and no one of the users need to change it? Add it to env variables and load config at bootstrap - exactly like in this homework.
* Occasionally - do users (e.g. admins) need to change the config dynamically? Save to a DB and expose an API endpoint to change the config.

And possibly more options which don't come to mind at the moment.

**Review by vpintaric-axilis — APPROVED** [review]

**Pull request merged**